plugins/hu.elte.txtuml.api.model/src/hu/elte/txtuml/api/model/ModelExecutorThread.java

```diff
@@ -76,24 +79,20 @@ private void newMailboxEntry(MailboxEntry entry) {
 	 * becomes empty.
 	 */
 	void shutdown() {
-		try {
-			class ShutdownAction implements MailboxEntry {
-				@Override
-				public void execute() {
-					if (mailbox.isEmpty()) {
-						ModelExecutor.shutdownNow();
-					} else {
-						try {
-							mailbox.put(new ShutdownAction());
-						} catch (InterruptedException e) {
-						}
-					}
-				}
-			}
+		shutdownWhenEmpty = true;

```

Comment:
If the ModelExecutorThread instance reads this variable, it can cache its value and reuse that cached value unless some kind of synchronization happens. If another thread changes the value of this variable, it may not affect the ModelExecutorThread in any way.